src/couch/src/couch_file.erl

```diff
@@ -462,7 +476,12 @@ handle_call({set_db_pid, Pid}, _From, #file{db_monitor=OldRef}=File) ->
     {reply, ok, File#file{db_monitor=Ref}};
 
 handle_call(sync, _From, #file{fd=Fd}=File) ->
-    {reply, file:sync(Fd), File};
+    case file:sync(Fd) of
+        ok ->
+            {reply, ok, File};
+        Error ->
+            {stop, Error, Error, File}
```

Comment:
any reason to keep `File` in state? save another accident by setting it to `nil` here?